### ISSUE TITLE
INFO: Definitive Ruleset Definition

### ISSUE THREAD

**Issue by ayomeer** [ayomeer/DNF_Tower#7]
[TowerOfDeath_Rules.md](https://github.com/ayomeer/DNF_Tower/blob/master/TowerOfDeath_Rules.md)

This is the version we're going to keep up to date and consider the master. Feel free to discuss in this issue thread. If you want to propose big changes, I recommend starting a separate Issue though to keep these nice and focused. ❤️

**Comment by Bumbadawg** [comment]
@ayomeer 
Some remarks :)
> Bo1 challenge

Only Bo1 ? :O

> 3.1 Players can become a floor master if they beat the current floor master in a BO5 Format. 

AND they cannot beat higher masters, otherwise Evil become master of Floor 1 :3

> 3.2 Admins need to be informed of this challenge and they will set up the BO5.

This can be automated by a server plugin. Both challengers type a command like !masterchallenge. If both agree upon, procedure is triggered. Any abandon/disconnect/forfeit cancels this procedure.

> The Map pool

every famous duel maps, but i think it'll be a pain in the ass for cointoss. Does the master choose like 4 in the pool for the Bo5 or else ?

Otherwise rules are pretty clear.

**Comment by ayomeer** [comment]
Yeah I think the BO1 thing isn't quite up to date. Only initial unranked game is BO1 now. I don't quite understand your Evil point. Do you mean, he'd be master of multiple floors? Naturally he'd only be master of the highest floor he beat.

**Comment by Bumbadawg** [comment]
Right, about my EVIL point, it's because i have in mind to automate this master mechanism in the server plugin and one player could become unfairly master while he could beat higher floors.
This would not happen with fair/regular players but could be a possible flaw to an automated server plugin.
Just keep that eventuality in mind.
As of now, masters will be named by almighty Admins.
But automating this system could benefit other players who'd like to reuse this idea/plugin on higher levels or different gametypes :-)

**Comment by aesick** [comment]
@Bumbadawg @ayomeer 

> Only Bo1

Yes. We are doing the rank games in a BO1 Format. Reason behind is that new players are focused to learn fewer maps at the beginning. Which should let them progress faster. Most of the knowledge can then be transfert to the next map.

> 3.1

You are right. A ranked player can only challenge the floor master he beat in a BO1. I edited the doc.

> 3.2 & Map pool

I'm not totaly sure that we should let the master system be run by a plugin. Masters should help the new players, that means we dont realy want any master who just do it as a prestige thing. Need to discuss this point a little futher.

**Comment by Bumbadawg** [comment]
There will be admin overriding for control, but automating it will help validate our rules.
Ok for BO1.
@aesick
